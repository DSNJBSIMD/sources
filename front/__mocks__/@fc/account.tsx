import type { PropsWithChildren } from 'react';
import React from 'react';

export const AccountContext = React.createContext(expect.any(Object));

export const AccountProvider = jest.fn(
  ({ children, config }: { config: unknown } & PropsWithChildren) => (
    <AccountContext.Provider value={config}>
      <div data-mockid="AccountProvider">
        <div>{children}</div>
      </div>
    </AccountContext.Provider>
  ),
);

export const ConnectValidator = {
  validate: jest.fn(),
};

export const Options = {
  CONFIG_NAME: 'Account',
};
