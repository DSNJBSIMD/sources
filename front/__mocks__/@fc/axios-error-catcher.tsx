import type { PropsWithChildren } from 'react';
import React from 'react';

export const AxiosErrorCatcherContext = React.createContext(expect.any(Object));

export const AxiosErrorCatcherProvider = jest.fn(
  ({ children, value }: { value: unknown } & PropsWithChildren) => (
    <AxiosErrorCatcherContext.Provider value={value}>{children}</AxiosErrorCatcherContext.Provider>
  ),
);
