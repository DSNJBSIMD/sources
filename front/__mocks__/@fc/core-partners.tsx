export const CreateInstanceButton = jest.fn(() => <div data-mockid="CreateInstanceButton" />);

export const InstanceComponent = jest.fn(() => <div data-mockid="InstanceComponent" />);

export const InstancesListComponent = jest.fn(() => <div data-mockid="InstancesListComponent" />);

export const PartnersService = {
  commit: jest.fn(),
  get: jest.fn(),
};

export enum Environment {
  SANDBOX = 'SANDBOX',
  PRODUCTION = 'PRODUCTION',
}

export enum Options {
  CONFIG_NAME = 'Partners',
}

export enum PublicationStatus {
  DRAFT = 'DRAFT',
  PENDING = 'PENDING',
  PUBLISHED = 'PUBLISHED',
}
