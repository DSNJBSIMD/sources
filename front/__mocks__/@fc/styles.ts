import type { ReactElement } from 'react';
import React from 'react';

import type { StylesProviderProps } from '@fc/styles';

export const useStylesQuery = jest.fn();

export const useStylesContext = jest.fn();

export const useStylesVariables = jest.fn(() => []);

export const StylesProvider = jest.fn(
  ({ children }: StylesProviderProps) => children as ReactElement,
);

export const StylesContext = React.createContext(expect.any(Object));

export enum Strings {
  DOUBLE_DASH = '--',
}
