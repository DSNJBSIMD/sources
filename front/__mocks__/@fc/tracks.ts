export const TracksListComponent = jest.fn();

export enum Options {
  CONFIG_NAME = 'Tracks',
}
