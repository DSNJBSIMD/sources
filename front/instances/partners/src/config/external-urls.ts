export const ExternalUrls: Record<string, string> = {
  spConfigurationDocUrl:
    'https://docs.partenaires.franceconnect.gouv.fr/fs/devenir-fs/projet-bac-a-sable/#configuration-de-votre-fournisseur-de-service',
};
