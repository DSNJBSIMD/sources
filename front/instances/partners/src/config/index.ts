import { Account } from './account';
import { DTO2Form } from './dto2form';
import { ExternalUrls } from './external-urls';
import { HttpClient } from './http-client';
import { Layout } from './layout';
import { Matomo } from './matomo';
import { Partners } from './partners';

export const AppConfig = {
  Account,
  DTO2Form,
  ExternalUrls,
  HttpClient,
  Layout,
  Matomo,
  Partners,
};
