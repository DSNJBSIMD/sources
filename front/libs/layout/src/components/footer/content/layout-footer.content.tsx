import React from 'react';

import { ConfigService } from '@fc/config';
import { LinkComponent } from '@fc/dsfr';

import { Options } from '../../../enums';
import type { LayoutConfig } from '../../../interfaces';

export const LayoutFooterContentComponent = React.memo(() => {
  const config = ConfigService.get<LayoutConfig>(Options.CONFIG_NAME);
  const { description, links } = config.footer;

  return (
    <div className="fr-footer__content">
      {description && <p className="fr-footer__content-desc">{description}</p>}
      {links && (
        <ul className="fr-footer__content-list">
          {links.map(({ external, href, label, title }, index) => {
            const uniqKey = `layout-footer-content-links::${index}`;
            return (
              <li key={uniqKey} className="fr-footer__content-item">
                <LinkComponent
                  className="fr-footer__content-link"
                  external={external}
                  href={href}
                  title={title}>
                  {label}
                </LinkComponent>
              </li>
            );
          })}
        </ul>
      )}
    </div>
  );
});

LayoutFooterContentComponent.displayName = 'LayoutFooterContentComponent';
