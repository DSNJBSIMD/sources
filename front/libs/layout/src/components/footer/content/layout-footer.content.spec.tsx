import { render } from '@testing-library/react';

import { ConfigService } from '@fc/config';
import { LinkComponent } from '@fc/dsfr';

import { LayoutFooterContentComponent } from './layout-footer.content';

describe('LayoutFooterContentComponent', () => {
  beforeEach(() => {
    // Given
    jest.mocked(ConfigService.get).mockReturnValue({
      footer: {
        description: undefined,
        links: undefined,
      },
    });
  });

  it('should call ConfigService.get with Layout config name', () => {
    // When
    render(<LayoutFooterContentComponent />);

    // Then
    expect(ConfigService.get).toHaveBeenCalledOnce();
    expect(ConfigService.get).toHaveBeenCalledWith('Layout');
  });

  it('should match the snapshot when description and links are defined', async () => {
    // Given
    jest.mocked(ConfigService.get).mockReturnValue({
      footer: {
        description: 'any description mock',
        links: [
          {
            href: 'any-href-mock-1',
            label: 'any-label-mock-1',
            title: 'any-title-mock-1',
          },
          {
            external: true,
            href: 'any-href-mock-2',
            label: 'any-label-mock-2',
            title: 'any-title-mock-2',
          },
        ],
      },
    });

    // When
    const { container, getByText } = render(<LayoutFooterContentComponent />);
    const descriptionElt = getByText('any description mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(descriptionElt).toBeInTheDocument();
    expect(LinkComponent).toHaveBeenCalledTimes(2);
    expect(LinkComponent).toHaveBeenNthCalledWith(
      1,
      {
        children: 'any-label-mock-1',
        className: 'fr-footer__content-link',
        external: undefined,
        href: 'any-href-mock-1',
        title: 'any-title-mock-1',
      },
      undefined,
    );
    expect(LinkComponent).toHaveBeenNthCalledWith(
      2,
      {
        children: 'any-label-mock-2',
        className: 'fr-footer__content-link',
        external: true,
        href: 'any-href-mock-2',
        title: 'any-title-mock-2',
      },
      undefined,
    );
  });
});
