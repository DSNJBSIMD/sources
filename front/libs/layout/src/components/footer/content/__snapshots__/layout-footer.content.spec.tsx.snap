// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutFooterContentComponent should match the snapshot when description and links are defined 1`] = `
<div>
  <div
    class="fr-footer__content"
  >
    <p
      class="fr-footer__content-desc"
    >
      any description mock
    </p>
    <ul
      class="fr-footer__content-list"
    >
      <li
        class="fr-footer__content-item"
      >
        <span
          data-mockid="LinkComponent"
        />
      </li>
      <li
        class="fr-footer__content-item"
      >
        <span
          data-mockid="LinkComponent"
        />
      </li>
    </ul>
  </div>
</div>
`;
